src/curator/toolforge.py

```
@@ -41,37 +68,116 @@ async def verify_api_key(x_api_key: str = Depends(X_API_KEY)):
+        func (Callable): The function to decorate.
 
     Returns:
-        Dict[str, Any]: The jobs data
+        Callable: The decorated function.
+    """
+    @functools.wraps(func)
+    async def wrapper(*args, **kwargs):
+        try:
+            return await func(*args, **kwargs)
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)